src/pytest_alembic/plugin/plugin.py

```diff
@@ -1,51 +1,124 @@
 import re
+from dataclasses import dataclass
+from typing import Callable, Dict, List, Optional
 
 import pytest
 from _pytest import fixtures
 
 from pytest_alembic.plugin.error import AlembicReprError, AlembicTestFailure
 
 
-def collect_all_tests():
-    from pytest_alembic import tests
+@dataclass(frozen=True)
+class PytestAlembicTest:
+    raw_name: str
+    function: Callable
+    is_experimental: bool
 
-    all_tests = {}
-    for name in dir(tests):
-        if name.startswith("test_"):
-            all_tests[name[5:]] = getattr(tests, name)
+    @property
+    def name(self):
+        # Chop off the "test_" prefix.
+        return self.raw_name[5:]
 
-    return all_tests
 
+@dataclass
+class _TestCollector:
+    available_tests: Dict[str, PytestAlembicTest]
 
-def parse_raw_test_names(raw_test_names):
-    test_names = re.split(r"[,\n]", raw_test_names)
+    included_tests: Optional[List[str]] = None
+    included_experimental_tests: Optional[List[str]] = None
+    excluded_tests: Optional[List[str]] = None
 
-    result = []
-    for test_name in test_names:
-        test_name = test_name.strip()
-        if not test_name:
-            continue
-        result.append(test_name)
-    return result
+    @classmethod
+    def collect(cls, default=True, experimental=True):
```

Comment:
Is this `default` arg used anywhere?

Reply:
They're both used/swapped in the helptext generation. The impl usage could probably be cleaner eventually, but it was kind of incidental to the actual feature